### PR TITLE
DIN-34 Add S3 transport

### PR DESCRIPTION
**Pull request by velppa**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 language: python
+sudo: required
+dist: xenial
```
Comment (velppa) [inline .travis.yml:3]: to get python 3.7 support - 
https://github.com/travis-ci/travis-ci/issues/9069#issuecomment-425720905

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 python:
-  - "3.3"
-  - "3.4"
   - "3.5"
+  - "3.6"
+  - "3.7"
```
Comment (germangh) [inline .travis.yml:7]: Is it worth testing in several Python versions? it will make travis run considerably slower.

File: analytics/client.py [hunk 3]
```diff
@@ -22,22 +23,37 @@
...
         self.write_key = write_key
         self.endpoint = endpoint
         self.on_error = on_error
         self.debug = debug
         self.send = send
 
+        if transport not in ('http', 's3'):
+            raise ValueError("transport should be either http or s3")
+
+        self.consumer = Consumer(self.queue, write_key, endpoint=endpoint,
```
Comment (germangh) [inline analytics/client.py]: A matter of taste, but isn't it more readable?

```
if transport == 'http':
    self.consumer = ...
elif transport == 's3':
    self.consumer = ...
else:
    raise ValueError(...)
```

File: README.md [hunk 4]
```diff
@@ -9,27 +8,87 @@ analytics-python is a python client is a slightly modified version of [Segment's
...
 
-The documentation for Segment's Python SDK that this repository is based on is available at [https://segment.com/libraries/python](https://segment.com/libraries/python). You can use Segment's docs to get familiar with the API.
+import analytics
 
+ANALYTICS_WRITE_KEY='AWS_API_GATEWAY_KEY'
+ANALYTICS_ENDPOINT='https://segment.fih.io/v1/[endpoint-key]'
 
-## License
+def log_error(e, batch):
+  print("exception: {}, batch: {}".format(e, batch), flush=True)
```
Comment (germangh) [inline README.md:40]: Better use `logging` for this.

**Review by germangh — APPROVED** [review]

File: analytics/client.py (hunk 3 above)
Comment (velppa) [reply analytics/client.py]: Agree!

File: README.md (hunk 4 above)
Comment (velppa) [reply README.md:40]: That's an example, one can use whatever is convenient. By default it `on_error` is `None` meaning SDK simply skipping errors.

**Pull request merged**